org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/BuilderParticipant.java

```diff
@@ -679,6 +680,7 @@ protected void saveResourceStorage(Resource resource, IFileSystemAccess access)
 		if (resource instanceof StorageAwareResource && access instanceof IFileSystemAccessExtension3) {
 			IResourceStorageFacade storageFacade = ((StorageAwareResource) resource).getResourceStorageFacade();
 			if (storageFacade != null) {
+				EcoreUtil2.resolveLazyCrossReferences(resource, CancelIndicator.NullImpl);
```

Comment:
will this have any consequences for the normal usecase? with the normal disabled case the rs never will be cleared
should we find out which clustering policy is used?

Reply:
Yes, that is true, but if the resource has not been cleared, EcoreUtil2.resolveLazyCrossReferences will not do anything more than what the builder already did. So it should not have consequences.

As a side note, I think the real fix would be to save the resource in the builder, and not in the builder participant, because then the problem does not exists to begin with, but I do not know why it is not like that already, and thus propose the smallest change possible.

Reply:
was this measured?

Reply:
maybe we can guard this with an injected boolean.